preview/Win7Msix/Win7MSIXInstaller/MsixRequest.cpp

```diff
@@ -55,110 +53,71 @@ std::map<PCWSTR, HandlerInfo> AddHandlers =
     return S_OK;
 }
 
-HRESULT MsixRequest::InitializeFilePathMappings()
-{
-    return m_filePathMappings.Initialize();
-}
-
 HRESULT MsixRequest::ProcessRequest()
 {
+    auto result = DeploymentResult();
+    result.Progress = 0;
+    result.Status = InstallationStep::InstallationStepStarted;
+    SendCallback(result);
+
     switch (m_operationType)
     {
-        case OperationType::Add:
-        {
-            RETURN_IF_FAILED(ProcessAddRequest());
-            break;
-        }
-        case OperationType::Remove:
-        {
-            RETURN_IF_FAILED(ProcessRemoveRequest());
-            break;
-        }
-        case OperationType::FindAllPackages:
-        {
-            RETURN_IF_FAILED(FindAllPackages());
-            break;
-        }
-        case OperationType::FindPackage:
-        {
-            RETURN_IF_FAILED(DisplayPackageInfo());
-            break;
-        }
-        default:
-            return HRESULT_FROM_WIN32(ERROR_NOT_SUPPORTED);
-    }
-
-    return S_OK;
-}
-
-HRESULT MsixRequest::DisplayPackageInfo()
-{
-    AutoPtr<IPackageHandler> handler;
-    RETURN_IF_FAILED(PopulatePackageInfo::CreateHandler(this, &handler));
-    HRESULT packageFoundResult = handler->ExecuteForRemoveRequest();
-
-    if (packageFoundResult == S_OK)
+    case OperationType::Add:
     {
-        std::wcout << std::endl;
-        std::wcout << L"PackageFullName: " << m_packageInfo->GetPackageFullName().c_str() << std::endl;
-        std::wcout << L"DisplayName: " << m_packageInfo->GetDisplayName().c_str() << std::endl;
-        std::wcout << L"DirectoryPath: " << m_packageInfo->GetPackageDirectoryPath().c_str() << std::endl;
-        std::wcout << std::endl;
+        RETURN_IF_FAILED(ProcessAddRequest());
+        break;
     }
-    else
+    case OperationType::Remove:
     {
-        std::wcout << std::endl;
-        std::wcout << L"Package not found " << std::endl;
+        RETURN_IF_FAILED(ProcessRemoveRequest());
+        break;
+    }
+    default:
+        return HRESULT_FROM_WIN32(ERROR_NOT_SUPPORTED);
     }
 
     return S_OK;
 }
 
-HRESULT MsixRequest::FindAllPackages()
+HRESULT MsixRequest::ProcessAddRequest()
 {
-    int numPackages = 0;
-    for (auto& p : std::experimental::filesystem::directory_iterator(m_filePathMappings.GetMsix7Directory()))
+    auto filemapping = FilePathMappings::GetInstance();
+    auto res = filemapping.GetInitializationResult();
+    if (FAILED(res))
     {
-        std::cout << p.path().filename() << std::endl;
-        numPackages++;
+        return E_FAIL;
     }
+    PackageInfo* packageInfo;
+    PopulatePackageInfo::GetPackageInfoFromPackage(this->m_packageFilePath.data(), this->m_validationOptions, &packageInfo);
 
-    std::cout << numPackages << " Packages found" << std::endl;
-    
-    return S_OK;
-}
+    if (packageInfo == nullptr)
+    {
+        return E_FAIL;
+    }
 
-HRESULT MsixRequest::ProcessAddRequest()
-{
-    PCWSTR currentHandlerName = PopulatePackageInfo::HandlerName;

```

Comment:
>PopulatePackageInfo [](start = 32, length = 19)

we wanted not to have free-floating work done outside of the handlers, which is why we created a PopulatePackageInfo handler to encapsulate that work.

Reply:
In this case, we should also include `PopulatePackageInfo` in `RemoveHandlers (the current version on master calls it before the loop), what's your opinion?